vars/dockerBuildArgs.groovy

```diff
@@ -16,6 +16,47 @@
    *
    * config['deps_build'] Whether to build the daos dependencies.
    *
+   * config['repo_type'] Type of repo to add.  Default 'local'
+   *
+   * Repositories URLs are looked up via Jenkins environment variables.
+   * There are two environment variables that are put together to create
+   * a path to a repo.
+   *
+   * The first envronment variable is "REPOSITORY_URL" which is a common
+   * base URL that is used for all accesses to the repository.
+   *
+   * The rest of the URL for each specific repo is in additional environment
+   * varables with a name format of "DAOS_STACK_$(distro}${mod}_${type}_REPO".
+   *
+   * The ${mod} currently is "_DOCKER" for repositories that are for use
+   * with dockerfiles, and "" for generic repositories.
+   * DOCKER repositories are for both replacing the distro built in
+   * repositories and adding locally built packages.
```

Comment:
What's the motivation for having Docker specific repos?  How/why are they different from generic repos?

Reply:
Dockerfiles need to have the option of replacing the distro provided repos with local repos and they also need locally built artifact repos added to them.

VMs/Physical nodes at our site are built with the replacement for distro provided repos already so only the locally built artifact repos need to be added.

By using a artifact server with and a group repository, we go from needing to supply two repos to just supplying one, and we also gain a limited ability to change the repository on a PR basis.

Reply:
> VMs/Physical nodes at our site are built with the replacement for distro provided repos already so only the locally built artifact repos need to be added.

Could Dockerfiles not mirror that?  Just seems like more work to keep separate CI image and Docker image repo definitions.